mlem/contrib/kubernetes/base.py

```diff
@@ -0,0 +1,281 @@
+    "Running": DeployStatus.RUNNING,
+    "Succeeded": DeployStatus.STOPPED,
+    "Failed": DeployStatus.CRASHED,
+    "Unknown": DeployStatus.UNKNOWN,
+}
+
+
+def find_index(nodes_list, node_name):
+    for i, each_node in enumerate(nodes_list):
+        if each_node.metadata.name == node_name:
+            return i
+    return -1
+
+
+class ServiceType(MlemABC):
+
+    abs_name: ClassVar = "k8s_service_type"
+    type: ClassVar = "kubernetes"
+
+    class Config:
+        type_root = True
+
+    def get_host_and_port(
+        self, service, namespace="mlem"  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[str], Optional[int]]:
+        """Returns host and port for the service in Kubernetes"""
+        return None, None
+
+
+class NodePortService(ServiceType):
+
+    type: ClassVar = "k8s_nodeport_service"
+
+    def get_host_and_port(self, service, namespace="mlem"):
+        host, port = None, None
+        port = service.items[0].spec.ports[0].node_port
+        node_name = (
+            client.CoreV1Api()
+            .list_namespaced_pod(namespace)
+            .items[0]
+            .spec.node_name
+        )
+        node_list = client.CoreV1Api().list_node().items
+        node_index = find_index(node_list, node_name)
+        address_dict = node_list[node_index].status.addresses
+        for each_address in address_dict:
+            if each_address.type == "ExternalDNS":
+                host = each_address.address
+        return host, port
+
+
+class LoadBalancerService(ServiceType):
+
+    type: ClassVar = "k8s_loadbalancer_service"
+
+    def get_host_and_port(self, service, namespace="mlem"):
+        host, port = None, None
+        port = service.items[0].spec.ports[0].port
+        ingress = service.items[0].status.load_balancer.ingress[0]
+        host = ingress.hostname or ingress.ip
+        return host, port
+
+
+class K8sDeploymentState(DeployState):
+    """DeployState implementation for Kubernetes deployments"""
+
+    type: ClassVar = "kubernetes"
+
+    image: Optional[DockerImage] = None
+    """Docker Image being used for Deployment"""
+    deployment_name: Optional[str] = None
+    """Name of Deployment"""
+
+
+class K8sDeployment(MlemDeployment, K8sYamlBuildArgs):
+    """MlemDeployment implementation for Kubernetes deployments"""
+
+    type: ClassVar = "kubernetes"
+    state_type: ClassVar = K8sDeploymentState
+    """type of state for Kubernetes deployments"""
+
+    server: Optional[Server] = None
+    """type of Server to use, with options such as FastAPI, RabbitMQ etc."""
+    registry: Optional[DockerRegistry] = DockerRegistry()
+    """docker registry"""
+    daemon: Optional[DockerDaemon] = DockerDaemon(host="")
+    """docker daemon"""
+    kube_config_file_path: Optional[str] = None
+    """path for kube config file of the cluster"""
+    templates_dir: List[str] = []
+    """list of dirs where templates reside"""
+
+    def load_kube_config(self):
+        config.load_kube_config(
```

Comment:
does this set config globally? we should not pollute users configuration if there is any. is there a way to not load this globally?

Reply:
As in? The config already is present in a user's system, this command just makes the K8s API aware of it. So we are not really polluting anything. We are just using what is already present. And with each new run of the python process / our CLI tool, config has to be loaded again and again. Different invocations won't remember that we already loaded it before.

Reply:
If user calls mlem api from his code while simultaneously working with `kubernetes` python package with configuration loaded from different source, you will pollute his configuration

Reply:
create an issue for that, we'll do it separately